### ISSUE TITLE
existing tools are removed from an agent with tools are updated 

### ISSUE THREAD

**Issue by sarahwooders** [cpacker/MemGPT#1828]
From a discord user: 

_When trying to create a new tool and add it to an already existing agent, the agent's tools are all wiped except for the added tool, I believe it is because tool adding is handled by the update_agent() function, which requires a list of tools to replace the original set of tools, I managed to mitigate this problem by getting the already added tools into this list as well as my new tool before updating it using that function using the python SDK. I believe this simple bugfix can also be applied to how the ADE behaves with tool handling by acquiring the agent's already added tools and adding them into the tools=[] list under the arguments for the update_agent() function._

**Comment by ketchalegend** [comment]
Also I will like to add: I noticed this bug yesterday. 

When you make changes to a tool using the ADE it seems to be saved. But doing a get request to the tools endpoint shows that the tool has the new code but the required fields are still are old ones from before the save. It seems like they never get updated until the tool is completely deleted and added again. 

I defined new required fields in my function but the ai kept trying to call the function with old fields which no longer existed. I had to delete the tool using the restapi completely and create it again.
